### PR TITLE
Search Bar Tests

### LINKED ISSUE #58: Search Bar Tests

### PR DESCRIPTION
**Pull request by lealgamb**
Added tests for the search bar. 

Fixes #58

### CODE REVIEW

File: src/test/java/codeu/controller/SearchServletTest.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    StringWriter stringWriter = new StringWriter();
+    PrintWriter writer = new PrintWriter(stringWriter);
+    Mockito.when(mockResponse.getWriter()).thenReturn(writer);
+
+    searchServlet.doGet(mockRequest, mockResponse);
+
+    Mockito.when(mockResponse.getContentType()).thenReturn("application/json");
+    Mockito.when(mockResponse.getCharacterEncoding()).thenReturn("UTF-8");
+
+    Assert.assertEquals("application/json", mockResponse.getContentType());
```
Comment (lealgamb) [inline src/test/java/codeu/controller/SearchServletTest.java]: I'm curious why lines 55-56 are needed for lines 58-59 to work? I still can't completely wrap my head around this.

Comment (vocatwk) [reply src/test/java/codeu/controller/SearchServletTest.java]: I'm not sure exactly why you're calling getContentType() and getCharacterEncoding() on the response object. If you want to verify that it calls the setter methods correctly with the correct arguments, you could just use argument captor.

File: src/test/java/codeu/controller/SearchServletTest.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    Assert.assertEquals("UTF-8", mockResponse.getCharacterEncoding());
+    Assert.assertEquals(json, stringWriter.toString());
```
Comment (vocatwk) [inline src/test/java/codeu/controller/SearchServletTest.java]: I'm also a little confused about this line. You created the json object and the stringWriter in this test method, what does this have to do with testing the doGet() method.

File: src/test/java/codeu/controller/SearchServletTest.java (hunk 1 above)
Comment (yebrahim) [reply src/test/java/codeu/controller/SearchServletTest.java]: It confused me for a while too, but once I found [this SO question](https://stackoverflow.com/questions/50403688/return-of-getcontenttype-in-mocked-httpservletresponse-instance-is-null) I thought, of course! You're not using an actual `HttpServletResponse` object here, so the `setContentType` call doesn't really do anything.

File: src/test/java/codeu/model/store/basic/UserStoreTest.java [hunk 3]
```diff
@@ -75,6 +75,33 @@ public void testGetUser_byId_notFound() {
...
 
+  @Test
+  public void testGetUsers_byUsername_found() {
+    List<User> resultUsers = userStore.getUsers("username");
+    Assert.assertEquals(3, resultUsers.size());
+    assertEquals(USER_ONE, resultUsers.get(0));
+    assertEquals(USER_TWO, resultUsers.get(1));
+    assertEquals(USER_THREE, resultUsers.get(2));
+  }
+
```
Comment (yebrahim) [inline src/test/java/codeu/model/store/basic/UserStoreTest.java]: How about one more test here to make sure partial strings are matched? For your sample users in this file, "USER_T" for example should return two results: "USER_TWO" and "USER_THREE".

File: src/test/java/codeu/controller/SearchServletTest.java (hunk 1 above)
Comment (lealgamb) [reply src/test/java/codeu/controller/SearchServletTest.java]: That link cleared things up for me, thanks. I tried using an argument captor but I understood the link example more.

File: src/test/java/codeu/controller/SearchServletTest.java (hunk 2 above)
Comment (lealgamb) [reply src/test/java/codeu/controller/SearchServletTest.java]: I included this to verify that the json is created properly.
